src/frontend/src/eth/components/transactions/Transaction.svelte

```diff
@@ -23,22 +38,84 @@
 	let timestamp: number | undefined;
 	let displayTimestamp: number | undefined;
 
+	let ckMinterInfoAddresses: (EthAddress | undefined)[];
+	$: $ckEthMinterInfoStore,
+		$ethereumToken,
+		(() => {
+			const ckMinterInfoData = $ckEthMinterInfoStore?.[$ethereumToken.id];
```

Comment:
This feels like it doesn't belong here.

- We do not want to map and iterate over the minter addresses on each transaction render for performance reasons.

- I understand now why you primarily created a global store, but even with such a store, we would still need to map here.

- However, the component should remain as naive as possible, and we are doing the opposite by adding logic here. "It's just rendering".

Spontaneously, I feel like we need something similar to IC side; we need some kind of mapper.

Basically:

sortedTransactions + ckMinterAddress.recude => map in a utils = upgraded list => <Transaction />

Do you know what I mean? What do you think?

Reply:
Ah, you mean to add a way to add the prop `type` directly to the transactions list?

Reply:
- Either the transaction component calls a map or the all combined list goes through a mapper.

- `ckMinterAddress.recude` should be performed only once.

Reply:
Maybe have a look at last commit, is that it?

Reply:
Yeah exactly, I think it's better. That way the component is naive and the heavy load is done has few times as possible without have global performance impact.

Can you add tests for the utils?

Reply:
Yes, makes sense!

I was waiting to create tests, until I confirmed this was the change

Reply:
mmmm you have broke something, a bunch of code you removed previously is back in the component

Reply:
nvm maybe I oversee something. let me have another look

Reply:
Added few minor questions as I had another look. Hope you don't mind. But this is good, thanks a ton!